src/main.ts

```diff
@@ -744,19 +788,75 @@ async function trackWindowState(browserWindow: BrowserWindow) {
 }
 
 function configureWatcher(browserWindow: BrowserWindow) {
-  watcher = watch([RECORDINGS_PATH, GENERATORS_PATH, SCRIPTS_PATH], {
+  watcher = watch([RECORDINGS_PATH, GENERATORS_PATH, SCRIPTS_PATH, DATA_PATH], {
     ignoreInitial: true,
   })
 
   watcher.on('add', (filePath) => {
-    browserWindow.webContents.send('ui:add-file', path.basename(filePath))
+    const file = getStudioFileFromPath(filePath)
+
+    if (!file) {
+      return
+    }
```

Comment:
Previously, you could add an arbitrary file to one of the watched folders and `ui:add-file` would still be fired